server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -698,57 +722,83 @@ private static ObjectMapper truncateObjectMapper(MapperMergeContext context, Obj
      *
      * @throws IllegalArgumentException if the mapper cannot be flattened
      */
-    List<FieldMapper> asFlattenedFieldMappers(MapperBuilderContext context) {
-        List<FieldMapper> flattenedMappers = new ArrayList<>();
+    List<Mapper> asFlattenedFieldMappers(MapperBuilderContext context, boolean throwOnFlattenableError) {
+        List<Mapper> flattenedMappers = new ArrayList<>();
         ContentPath path = new ContentPath();
-        asFlattenedFieldMappers(context, flattenedMappers, path);
+        asFlattenedFieldMappers(context, flattenedMappers, path, throwOnFlattenableError);
         return flattenedMappers;
     }
 
-    private void asFlattenedFieldMappers(MapperBuilderContext context, List<FieldMapper> flattenedMappers, ContentPath path) {
-        ensureFlattenable(context, path);
+    private static boolean isFlattenable(Optional<Subobjects> subobjects, ObjectMapper mapper, MapperBuilderContext context) {
```

Comment:
Maybe consider renaming? Seems that this doesn't return whether an ObjectMapper can be flattened (`isFlattenable`) but whether we should attempt to flatten it.

Reply:
I hope it's better now.